lib/parallel_tests/grouper.rb

```diff
@@ -15,13 +15,25 @@ def in_even_groups_by_size(items, num_groups, options= {})
         groups = Array.new(num_groups) { {:items => [], :size => 0} }
 
         # add all files that should run in a single process to one group
-        (options[:single_process] || []).each do |pattern|
-          matched, items = items.partition { |item, _size| item =~ pattern }
-          matched.each { |item, size| add_to_group(groups.first, item, size) }
+        single_process_patterns = options[:single_process] || []
+
+        single_items, items = items.partition do |item, _size|
+          single_process_patterns.any? { |pattern| item =~ pattern }
         end
 
-        groups_to_fill = (options[:isolate] ? groups[1..-1] : groups)
-        group_features_by_size(items_to_group(items), groups_to_fill)
+        isolated_count = options[:isolate] ? (options[:isolated_count] || 1) : 0
+
+        if isolated_count >= 1
```

Comment:
prefer to use `if options[:isolate]` here so it's clearer that this block is about isolation